Add alerts and dashboards

kakkoyun opened the pull request:
cc @squat

squat commented:
Can you re-commit this without adding the jsonnet/vendor dir? normally we expect that to be generated by jb with the lockfile, so there's no need to commit it to the repo

squat commented on Makefile:
```diff
@@ -0,0 +1,28 @@
+
+.PHONY: generate
+generate: alerts.yaml rules.yaml dashboards
+
```
there seems to be no makefile rule for these targets

squat commented on jsonnet/Makefile:
```diff
@@ -0,0 +1,58 @@
+EXAMPLES := ../examples
```
let's merge this Makefile with the one at the root so we only have one

squat commented on jsonnet/.gitignore:
```diff
@@ -0,0 +1 @@
+vendor
```
it should be sufficient to have the entry in the .gitignore at the root, no?

kakkoyun replied:
Definitely, I've forgotten to clean things up

kakkoyun commented:
cc @metalmatze

brancz commented:
Why are we committing a lockfile here at all?

squat commented:
ah, frederic means that since this repo doesnt actually commit any generated manifests, i.e. it is only used as a library that is imported by other jsonnet projects, it doesn't need to lock and dependencies.

kakkoyun commented:
@brancz for no particular reason, I didn't know we shouldn't. I'll remove it.

kakkoyun commented:
@squat Got that now 👍 Thanks for the explanation :)

brancz commented:
lgtm, leaving last call to @squat @metalmatze

squat commented on Makefile:
```diff
@@ -0,0 +1,70 @@
+all: generate fmt thanos-receive-controller
+
+thanos-receive-controller: main.go
+	go build -v
```
We should make sure this uses the same flags and environment variables as we have today in the build section of the drone.yml. Then, we should change the command in drone.yml to instead run `make thanos-receive-controller`

squat commented on Makefile:
```diff
@@ -0,0 +1,70 @@
+	go mod vendor
+
+.PHONY: fmt
+fmt:
```
drone.yml should invoke this rather than it's own script in the fmt section.

squat commented on Makefile:
```diff
@@ -0,0 +1,70 @@
+	echo ${JSONNET_SRC} | xargs -n 1 -- $(JSONNET_FMT) -i
+
+.PHONY: lint
+lint: fmt ${ALERTS} ${RULES}
```
we should add this to CI

squat commented on Makefile:
```diff
@@ -0,0 +1,70 @@
+	promtool check rules ${ALERTS} ${RULES}
+
+.PHONY: test ${ALERTS} ${RULES}
+test:
```
ensure `make test` is invoked by ci in the test section. we'll need to use an docker image that includes promtool.

squat commented on Makefile:
```diff
@@ -0,0 +1,70 @@
+	go build -v
+
+.PHONY: generate
+generate: ${ALERTS} ${RULES} ${DASHBOARDS}
```
Our CI should have a `generate` step that runs `make generate` and then ensures that `git diff --exit-code` exits clean.

kakkoyun commented:
@squat I have fixed emphasized make issues. I introduced two new docker images as drone stages
https://github.com/kakkoyun/docker-go-promtool and https://github.com/kakkoyun/docker-go-jsonnet-ci. I can put them some other common registry, I've just put them in my personal account for now.

squat commented:
To simplify the maintenance of docker containers, I think we should use the `quay.io/coreos/jsonnet-ci` container. We use this in several other projects (prometheus-operator, CMO, telemeter etc). The only thing it is missing is promtool, so we'd have to add that. This way you don't need to build and keep the other two containers up to date. WDYT?

kakkoyun commented:
@squat Totally makes sense. I'll send a PR to add `promtool`. I guess this is the place https://github.com/coreos/prometheus-operator/blob/master/scripts/jsonnet/Dockerfile

brancz commented:
We can move forward here now, the image is about to be published.

kakkoyun commented:
@squat `quay.io/coreos/jsonnet-ci` is updated as we talked https://github.com/coreos/prometheus-operator/pull/2739. It's merged and this PR updated. Ready for your review.

brancz commented:
lgtm giving @squat last call
